test/geometry/LinearPropagator.test.cc

```diff
@@ -79,59 +81,69 @@ TEST_F(LinearPropagatorHostTest, track_line)
 
     {
         // Track from outside detector, moving right
-        geo = {{-6, 0, 0}, {1, 0, 0}};
-        EXPECT_EQ(VolumeId{1}, geo.volume_id()); // World
+        geo = {{-10, 10, 10}, {1, 0, 0}};
+        EXPECT_EQ(VolumeId{0}, geo.volume_id()); // Shape2 center
 
         geo.find_next_step();
-        EXPECT_SOFT_EQ(1.0, geo.next_step());
+        EXPECT_SOFT_EQ(5, geo.next_step());
         propagate();
-        EXPECT_SOFT_EQ(-5.0, geo.pos()[0]);
-        EXPECT_EQ(VolumeId{0}, geo.volume_id()); // Detector
+        EXPECT_SOFT_EQ(-5, geo.pos()[0]);
+        EXPECT_EQ(VolumeId{1}, geo.volume_id()); // Shape2 -> Shape1
 
         geo.find_next_step();
-        EXPECT_SOFT_EQ(10.0, geo.next_step());
+        EXPECT_SOFT_EQ(1.0, geo.next_step());
         propagate();
-        EXPECT_EQ(VolumeId{1}, geo.volume_id()); // World
+        EXPECT_EQ(VolumeId{3}, geo.volume_id()); // Shape1 -> Envelope
         EXPECT_EQ(false, geo.is_outside());
 
         geo.find_next_step();
-        EXPECT_SOFT_EQ(45.0, geo.next_step());
+        EXPECT_SOFT_EQ(1.0, geo.next_step());
         propagate();
-        EXPECT_EQ(true, geo.is_outside());
+        EXPECT_EQ(false, geo.is_outside()); // leaving World
     }
 
     {
         // Track from outside edge fails
-        geo = {{50, 0, 0}, {-1, 0, 0}};
+        geo = {{24, 0, 0}, {-1, 0, 0}};
         EXPECT_EQ(true, geo.is_outside());
     }
 
     {
         // But it works when you move juuust inside
-        geo = {{50 - 1e-6, 0, 0}, {-1, 0, 0}};
+        real_type eps = 1e-6;
+        geo           = {{-24 + eps, 6.5, 6.5}, {1, 0, 0}};
         EXPECT_EQ(false, geo.is_outside());
-        EXPECT_EQ(VolumeId{1}, geo.volume_id()); // World
+        EXPECT_EQ(VolumeId{10}, geo.volume_id()); // World
         geo.find_next_step();
-        EXPECT_SOFT_EQ(45.0 - 1e-6, geo.next_step());
+        EXPECT_SOFT_EQ(7. - eps, geo.next_step());
         propagate();
-        EXPECT_EQ(VolumeId{0}, geo.volume_id()); // Detector
+        EXPECT_EQ(VolumeId{3}, geo.volume_id()); // World -> Envelope
+
+        // debugging
+        geo.volume().Print();
     }
     {
         // Track from inside detector
-        geo = {{0, 0, 0}, {1, 0, 0}};
-        EXPECT_EQ(VolumeId{0}, geo.volume_id()); // Detector
+        geo = {{-10, 10, 10}, {0, -1, 0}};
+        EXPECT_EQ(VolumeId{0}, geo.volume_id()); // Shape1 center
+
+        // info printout
+        std::cerr << "\n === current position: " << geo.pos() << "\n";
+        // geo.volume().Print();
```

Comment:
Debug code was left in here and a couple others; please remove them.

Reply:
Done.